src/javascript/app/pages/user/account/settings/two_factor_authentication.js

```diff
@@ -0,0 +1,116 @@
+const QRCode           = require('davidshimjs-qrcodejs');
+const Client           = require('../../../../base/client');
+const BinarySocket     = require('../../../../base/socket');
+const FormManager      = require('../../../../common/form_manager');
+const getPropertyValue = require('../../../../../_common/utility').getPropertyValue;
+const localize         = require('../../../../../_common/localize').localize;
+
+const TwoFactorAuthentication = (() => {
+    const form_id = '#frm_two_factor_auth';
+    const state = ['disabled', 'enabled'];
+    let current_state,
+        next_state,
+        qrcode; // eslint-disable-line
+
+    const onLoad = () => {
+        init();
+    };
+
+    const init = () => {
+        BinarySocket.send({ account_security: 1, totp_action: 'status'}).then((res) => {
+            $('#two_factor_loading').remove();
+
+            if (res.error) {
+                handleError('status', res.error.message);
+                return;
+            }
+
+            current_state = state[res.account_security.totp.is_enabled];
+            next_state    = state[+(!res.account_security.totp.is_enabled)].slice(0, -1);
+
+            $(`#${current_state}`).setVisibility(1);
+            $('#btn_submit').text(localize(next_state));
+            $(form_id).setVisibility(1);
```

Comment:
how about storing all the jquery elements we need onLoad of page once, and then we can use those variables instead of getting the element every time?

Reply:
That's a good idea, will update it